### PR TITLE
Add disabling of SMB1 to hardening cookbook

### PR DESCRIPTION
**Pull request by yvovandoorn**
* Add attribute to disable remediation of SMB1 
* Add inspec test to look for attributes

### CODE REVIEW

**Comment by yvovandoorn** [comment]
It looks like appveyor is having an issue getting some certs. /cc @grdnrio

File: test/integration/default/inspec/controls/tests.rb [hunk 1]
```diff
@@ -4,3 +4,15 @@
   skip_control 'cis-network-access-2.2.2'
   skip_control 'windows-account-100'
 end
+
+control "control_smb1_is_disabled" do
```
Comment (artem-sidorenko) [inline test/integration/default/inspec/controls/tests.rb]: should we be a part of [windows-baseline](https://github.com/dev-sec/windows-baseline)?

File: recipes/access.rb [hunk 2]
```diff
@@ -60,3 +60,16 @@
   }]
   action :create
 end
+
+return unless node['platform_family'] == 'windows'
```
Comment (artem-sidorenko) [inline recipes/access.rb]: I guess this is a left over from copy & paste :-)

File: recipes/access.rb [hunk 3]
```diff
@@ -60,3 +60,16 @@
...
+
+if node['windows_hardening']['smbv1']['disable'] == true
```
Comment (artem-sidorenko) [inline recipes/access.rb]: isn't [not_if](https://docs.chef.io/resource_common.html#not-if-examples) here a better option? I remember foodcritic complains about such ifs and suggests not_if/only_if

Comment (yvovandoorn) [inline recipes/access.rb]: Hey @artem-sidorenko! So I thought long and hard about this. FC023 is no longer enforced by food-critic (modern versions) and I copied the style used in https://github.com/dev-sec/chef-windows-hardening/blob/master/recipes/rdp.rb#L9.

File: recipes/access.rb (hunk 2 above)
Comment (yvovandoorn) [inline recipes/access.rb]: Yes! Fixing.

File: recipes/access.rb (hunk 3 above)
Comment (artem-sidorenko) [inline recipes/access.rb]: I did not know that :) Makes sense

File: test/integration/default/inspec/controls/tests.rb (hunk 1 above)
Comment (yvovandoorn) [inline test/integration/default/inspec/controls/tests.rb]: I'm ok with either. I'll go by what is the preferential way.

Comment (chris-rock) [inline test/integration/default/inspec/controls/tests.rb]: Agree with @artem-sidorenko  This test should move to windows baseline

**Comment by yvovandoorn** [comment]
@artem-sidorenko @chris-rock moved to https://github.com/dev-sec/windows-baseline/pull/16

**Comment by artem-sidorenko** [comment]
@chris-rock as far I remember the SMB1 was used for distribution of WannaCry. Maybe its a good idea to push a new release and some tweets when this PR is merged

**Comment by chris-rock** [comment]
I just try to get this green, but this is not related to this PR. @yvovandoorn Can you squash and sign-off?

**Comment by chris-rock** [comment]
@yvovandoorn #28 is in now, therefore this PR should turn green

**Comment by yvovandoorn** [comment]
@chris-rock just squashed the commits. Waiting for appveyor (fixed in your #28) to finish.

**Comment by chris-rock** [comment]
@yvovandoorn Do you want to write a blog post for this at https://github.com/dev-sec/blog/tree/gh-pages/_posts?

**Comment by yvovandoorn** [comment]
Sure thing! I'll write something up tonight / tomorrow morning.

**Pull request merged**